modules/wordads/wordads.php

```diff
@@ -206,15 +232,19 @@ private function insert_adcode() {
 		}
 
 		if ( $this->option( 'enable_header_ad', true ) ) {
-			switch ( get_stylesheet() ) {
-				case 'twentyseventeen':
-				case 'twentyfifteen':
-				case 'twentyfourteen':
-					add_action( 'wp_footer', array( $this, 'insert_header_ad_special' ) );
-					break;
-				default:
-					add_action( 'wp_head', array( $this, 'insert_header_ad' ), 100 );
-					break;
+			if ( self::is_amp() ) {
```

Comment:
To eliminate the `_doing_it_wrong` error, this cannot be called here since it  is before the `wp` action fires. So what you need to do here is move the `is_amp` check _inside_ the `the_content` filter, because that happens after `wp`. An easy thing to fix this would be just to defer the `init` method to the `wp` action, rather then where it is being called now.

Reply:
In other words, try:

```diff
--- a/modules/wordads/wordads.php
+++ b/modules/wordads/wordads.php
@@ -106,7 +106,7 @@ class WordAds {
 	 * @since 4.5.0
 	 */
 	function __construct() {
-		add_action( 'init', array( $this, 'init' ) );
+		add_action( 'wp', array( $this, 'init' ) );
 	}
 
 	/**
```

Reply:
@dereksmart do you foresee any issues for non-AMP output if I change the action from `init` to `wp`?

Reply:
🤔 That's a good question, I don't _think_ it'll be a problem, but I haven't done any testing to verify. You'll have to work with @rclations and @cphilleo to make sure everything is working as intended on this one.

Reply:
Tested a sample site, and it doesn't appear to effect ad rendering or ads.txt output - should be g2g 👍

Reply:
Change made in https://github.com/Automattic/jetpack/pull/12733/commits/65ae218f9ad439dc43fc8a3fa3dd9acabfe110c9